plugins/fabric8-multi-tenant-manager/src/main/java/com/redhat/che/multitenant/GuessedSubject.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2016-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package com.redhat.che.multitenant;
+
+import org.eclipse.che.api.core.ForbiddenException;
+import org.eclipse.che.commons.subject.Subject;
+
+public class GuessedSubject implements Subject {
```

Comment:
@davidfestal could you please add javadoc with explanation for this class and when it is supposed to be used ?

Reply:
Is it OK for you ?